durable-storage/src/merkle_layer/node.rs

```diff
@@ -0,0 +1,424 @@
+// SPDX-FileCopyrightText: 2025 Trilitech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+use std::cmp::Ordering;
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use bytes::Bytes;
+
+use super::Key;
+
+/// A node that supports rebalancing and Merklisation.
+#[derive(Clone, Default, Debug)]
+pub(super) struct MavlNode {
+    key: Key,
+    data: Bytes,
+    left: Option<Arc<Self>>,
+    right: Option<Arc<Self>>,
+
+    /// A [None] hash is a hash that has not been set or has been dirtied.
+    hash: Option<blake3::Hash>,
+
+    /// The difference in heights between child branches.
```

Comment:
```suggestion
    /// The difference in heights between child branches. (right - left child)
```

This now should be the convention all the time?